src/test/java/org/kiwiproject/dropwizard/util/job/MonitoredJobTest.java

```diff
@@ -0,0 +1,194 @@
+                    .isInstanceOf(IllegalArgumentException.class)
+                    .hasMessage("task is required");
+        }
+
+        @Test
+        void shouldSetDefaults() {
+            var job = MonitoredJob.builder()
+                    .name("Name and Task Job")
+                    .task(() -> System.out.println("Hello"))
+                    .build();
+
+            assertThat(job.getDecisionFunction()).isNotNull();
+            assertThat(job.getEnvironment()).isNotNull();
+        }
+    }
+
+    @Nested
+    class Run {
+
+        @Test
+        void shouldRunSyncWithoutErrorWhenActive(SoftAssertions softly) {
+            var environment = mock(KiwiEnvironment.class);
+            var mockedTime = System.currentTimeMillis();
+            when(environment.currentTimeMillis())
+                    .thenReturn(mockedTime)
+                    .thenReturn(mockedTime + 1)
+                    .thenReturn(mockedTime + 2);
+
+            var taskRunCount = new AtomicInteger();
+            var job = MonitoredJob.builder()
+                    .name("Run active sync no errors")
+                    .task(taskRunCount::incrementAndGet)
+                    .environment(environment)
+                    .build();
+
+            job.run();
+
+            softly.assertThat(taskRunCount.get()).isOne();
+            softly.assertThat(job.getLastExecutionTime().get()).isOne();
+            softly.assertThat(job.getLastSuccess().get()).isEqualTo(mockedTime + 2);
+            softly.assertThat(job.getLastFailure().get()).isZero();
+            softly.assertThat(job.getFailureCount().get()).isZero();
+        }
+
+        @Test
+        void shouldRunAsyncWithoutErrorWhenActiveAndTimeoutIsSet(SoftAssertions softly) {
+            var environment = mock(KiwiEnvironment.class);
+            var mockedTime = System.currentTimeMillis();
+            when(environment.currentTimeMillis())
+                    .thenReturn(mockedTime)
+                    .thenReturn(mockedTime + 1)
+                    .thenReturn(mockedTime + 2);
+
+            var taskRunCount = new AtomicInteger();
+            var job = MonitoredJob.builder()
+                    .name("Run active async no errors")
+                    .task(taskRunCount::incrementAndGet)
+                    .environment(environment)
+                    .timeout(Duration.ofSeconds(1))
+                    .build();
+
+            job.run();
+            await().atMost(Duration.ofSeconds(2)).until(() -> taskRunCount.get() > 0);
+
+            softly.assertThat(taskRunCount.get()).isOne();
+            softly.assertThat(job.getLastExecutionTime().get()).isOne();
+            softly.assertThat(job.getLastSuccess().get()).isEqualTo(mockedTime + 2);
+            softly.assertThat(job.getLastFailure().get()).isZero();
+            softly.assertThat(job.getFailureCount().get()).isZero();
+        }
+
+        @Test
+        void shouldSkipExecutionWhenNotActive(SoftAssertions softly) {
```

Comment:
Here's how you can change this test to try decision functions that return `false` and `null`:

```java
        @ParameterizedTest
        @MethodSource("org.kiwiproject.dropwizard.util.job.MonitoredJobTest#falseAndNullDecisionFunctions")
        void shouldSkipExecutionWhenDecisionFunctionReturnsFalseOrNull(
                Function<MonitoredJob<Runnable>, Boolean> decisionFun, SoftAssertions softly) {

            var environment = mock(KiwiEnvironment.class);

            // ...
```

And then add this method _outside_ the nested `Run` class:

```java
    static Stream<Function<MonitoredJob<Runnable>, Boolean>> falseAndNullDecisionFunctions() {
        return Stream.of(
                job -> false,
                job -> null
        );
    }
```